### PR TITLE
Send increase allowance for non native swaps

### PR DESCRIPTION
**Pull request by sashimi36**

### CODE REVIEW

File: features/swap/hooks/useTokenSwap.tsx [hunk 1]
```diff
@@ -76,27 +76,26 @@ export const useTokenSwap = ({
...
-          nativeAmount: convertedTokenAmount,
+        return await swapTokenAForTokenB({
+          tokenAmount: convertedTokenAmount,
           price: convertedPrice,
           slippage,
           senderAddress: address,
           swapAddress: streamlinePoolAB.swap_address,
-          tokenDenom: tokenA.denom,
+          tokenA,
+          tokenB,
```
Comment (ben2x4) [inline features/swap/hooks/useTokenSwap.tsx]: why do we need info on tokenB here?

File: services/swap/swapTokenForToken.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  price: number
+  slippage: number
+  senderAddress: string
+  swapAddress: string
+  outputSwapAddress: string
+  tokenA: TokenInfo
+  client: SigningCosmWasmClient
+}
+
+export const swapTokenForToken = async ({
```
Comment (ben2x4) [inline services/swap/swapTokenForToken.ts:18]: So I think this needs to be more flexible, the current code only lets you swap Token A. When you want to be able to swap tokenA or TokenB.

Comment (ben2x4) [reply services/swap/swapTokenForToken.ts:18]: technically I think it only worker with TokenB even though we use TokenA in the function signature.

File: services/swap/swapTokenForToken.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    pass_through_swap: {
+      output_min_token: `${minOutputToken}`,
+      input_token: 'Token2',
+      input_token_amount: `${tokenAmount}`,
+      output_amm_address: outputSwapAddress,
+    },
+  }
+
+  if (!tokenA.native) {
+    return executeSwapWithIncreasedAllowance({
```
Comment (ben2x4) [inline services/swap/swapTokenForToken.ts:41]: this needs a little more thinking with regards to gas fees. If we are doing a `pass_through_swap` we will need more gas then just a regular swap with an allowance message. we might want to pass the gas fee as an input to this function.

**Review by ben2x4 — CHANGES_REQUESTED** [review]
Making good progress here, added a few thoughts and suggestions.

File: features/swap/hooks/useTokenSwap.tsx (hunk 1 above)
Comment (sashimi36) [reply features/swap/hooks/useTokenSwap.tsx]: good call

**Review by ben2x4 — APPROVED** [review]
LGTM

**Pull request merged**